app/ts/components/requirements.tsx

```diff
@@ -197,7 +234,22 @@ export const Create = ( { contentHashInput, resolutionAddressInput, loadingInfos
 				<p class = 'error-component' style = 'width: 100%; margin-left: 10px; text-align: center;'> PetalLock contract is not deployed. </p>
 				<button class = 'button is-primary' onClick = { deploy }> Deploy PetalLock contract</button>
 			</> : <></> }
-			{ immutable.value ? <></> : <>
+			{ immutable.value ? <div class = 'extend-dialog'>
+				<p style = 'white-space: nowrap; margin: 0; font-size: 24px; padding-bottom: 10px; justify-self: center;'>{ `Renew ${ subDomain }` }</p>
+				<div style = 'justify-content: center;'>
+					<p style = 'font-size: 24px;'> Renew by&nbsp;</p> <YearPicker year = { extendYear }/> <p style = 'font-size: 24px;'>&nbsp;years </p>
+					<button style = 'font-size: 3em;' class = 'button is-primary' disabled = { extending.value } onClick = { renewByYear }> Renew { extending.value ? <Spinner/> : <></> }</button>
+				</div>
+				{ !twoLevelDomainExpiryBiggerThanLowestLevelExpiry.value || checkBoxes.deepValue[0] === undefined ? <></> : <>
+					<div style = 'justify-content: center; font-size: 24px;'>
+						<p> OR </p>
+					</div>
+					<div style = 'justify-content: center;'>
+						<p style = 'font-size: 24px;' >{ `Renew without renewing ${ checkBoxes.deepValue[0]?.domainInfo.subDomain }` }</p>
+						<button style = 'font-size: 3em;' class = 'button is-primary' disabled = { extending.value } onClick = { renewToMax }> { `Renew to ${ new Date(Number(checkBoxes.deepValue[0].domainInfo.expiry) * 1000).toISOString().substring(0, 10) }`} { extending.value ? <Spinner/> : <></> }</button>
```

Comment:
Is it impossible for `checkBoxes.deepValue[0].domainInfo.expiry` to be undefined or not a stringified number?  If something throws in here, I kinda suspect it will be hard to debug.

Reply:
it cannot be undefined and its a bigint, not a string. I don't think it can throw here. The number can be too big, but I don't think it would throw in that case

Reply:
I guess we could convert it to number before so this casting is not needed